hubble/cmd/observe/flows_filter.go

```diff
@@ -700,11 +698,19 @@ func (of *flowFilter) set(f *filterTracker, name, val string, track bool) error
 			f.NodeLabels = append(f.GetNodeLabels(), val)
 		})
 
-		// cluster Name filters
+	// cluster name filters
 	case "cluster":
 		f.apply(func(f *flowpb.FlowFilter) {
 			f.NodeName = append(f.GetNodeName(), val+"/")
 		})
+	case "from-cluster":
+		f.apply(func(f *flowpb.FlowFilter) {
+			f.SourceClusterName = append(f.GetSourceClusterName(), val)
+		})
+	case "to-cluster":
+		f.apply(func(f *flowpb.FlowFilter) {
+			f.DestinationClusterName = append(f.GetDestinationClusterName(), val)
+		})
```

Comment:
Hm. So this is interesting: `--cluster` checks the node name, where as `--from/to-cluster` checks the endpoint data. The two have different semantics.

With this implementation, `--cluster foo` will only give you flows observed by nodes in foo. `--from-cluster foo || --to-cluster foo` however gives you flows going between the clusters, including flows observed by nodes in other clusters than foo (but going or coming from cluster foo).

This is different from other flags. For example, if one uses `--namespace bar` it is equivalent with `--from-namespace foo OR --to-namespace foo`.

I think we should do a breaking change here and re-define how `--cluster` works, so it is consistent with all other flags. The old behavior is still available via `--node-name foo/`. We can document that.

Reply:
@gandro yes all make sense to me, I was trying to avoid breaking as part of this PR but you are correct in that it's bringing inconsistency. Related question: do we start to document Hubble CLI changes in the Cilium upgrade guide?

Reply:
> Related question: do we start to document Hubble CLI changes in the Cilium upgrade guide?

Yeah, I think adding a new section to the upgrade notes makes sense to me. We probably also want to add it to the notes on the Hubble release in `cilium/hubble`, but than needs to be done by the release managers once Hubble CLI 1.16 ships

Reply:
Thanks for the input, I've added a commit on top refactoring the `--cluster` flag as discussed, adding a note in the upgrade guide and (hopefully) summarized this discussion in the commit message. Please take another look :pray:

Reply:
I'm a bit torn on this. 

I agree that this new behaviour makes more sense and I'm oke with redefining how `--cluster` works.
What I don't particularly like is that `--cluster` will just stop working if you use the newly released Hubble CLI on an older cluster that doesn't support the new filter..

I don't have great idea to avoid this and I think in the end it makes more sense to have the flag this way. So I'm fine with keeping it as is, we just need to be aware of that.

Reply:
I've suggested we introduce some kind of version indicator previously. Eg: we could send a gRPC metadata header to clients when they connect, and the clients could use that to determine how they should behave.

Reply:
I haven't since we discussed it in person and deemed it unnecessary/extra complexity in those cases at least.

Reply:
Created https://github.com/cilium/cilium/issues/35724 to keep track of this with @chancez suggestion of sending the server version through gRPC metadata. Let me know if I missed something :pray: